File: src_20170503/src/web/server/fabs/views/position.py
# -*- coding: utf-8 -*-
"""
    yinsho.api.nets
    #####################

    yinsho performance view module
"""

from flask import Blueprint, request, json, abort, current_app, jsonify, session as web_session, g

from ..services import poConService
from . import route


bp = Blueprint('position', __name__, url_prefix='/position')
poService = poConService()


@route(bp, '/group_delete',methods=['POST'])
def group_delete():
    return poService.group_delete(**request.json)

@route(bp, '/department_delete',methods=['POST'])
def department_delete():
    return poService.department_delete(**request.json)

@route(bp, '/add',methods=['POST'])
def add():
    return poService.add(**request.json)

@route(bp, '/type_add',methods=['POST'])
def type_add():
    return poService.type_add(**request.json)

@route(bp, '/department_add',methods=['POST'])
def department_add():
    return poService.department_add(**request.json)

@route(bp, '/groups',methods=['POST'])
def groups():
    return poService.groups()

@route(bp, '/find',methods=['POST'])
def find():
    return poService.find(**request.json)

@route(bp, '/targets',methods=['POST'])
def targets():
    return poService.targets(**request.json)

@route(bp, '/objects',methods=['POST'])
def objects():
    return poService.objects()

@route(bp, '/save',methods=['POST'])
def save():
    return poService.save()

@route(bp, '/persons',methods=['POST'])
def persons():
    return poService.persons()

@route(bp, '/load',methods=['POST'])
def load():
    return poService.load()
	
@route(bp, '/add_save',methods=['POST'])
def add_save():
    return poService.add_save(**request.json)	

@route(bp, '/edit_save',methods=['POST'])
def edit_save():
    return poService.edit_save(**request.json)

@route(bp, '/group_edit_save',methods=['POST'])
def group_edit_save():
    return poService.group_edit_save(**request.json)

@route(bp, '/department_edit_save',methods=['POST'])
def department_edit_save():
    return poService.department_edit_save(**request.json)

File: src_20170503/src/web/web/fabs/services/position.js
/**
 * Position Service
 */
ysp.service('poConService', poConService);
poConService.$inject = ['$http'];
function poConService($http) {
        return {

        groups: function () {
            return $http.post(base_url + '/position/groups');
        },
        department_save:function(data){
            return $http.post(base_url + '/position/department_add',data);
        },
        group_save:function(data){
            return $http.post(base_url + '/position/add',data);
        },
        type_add:function(data){
            return $http.post(base_url + '/position/type_add',data);
        },
        group_delete:function(data){
            return $http.post(base_url + '/position/group_delete',data);
        },
        department_delete:function(data){
            return $http.post(base_url + '/position/department_delete',data);
        },
        group_edit_save:function(data){
            return $http.post(base_url + '/position/group_edit_save',data);
        },
        department_edit_save:function(data){
            return $http.post(base_url + '/position/department_edit_save',data);
        },
        check_groups: function (data) {
            console.log(data)
            return $http.post(base_url + '/position/check_groups', data);
        },
        ords: function (data) {
            return $http.post(base_url + '/position/ords', data);
        }

    };

       
    
};

